test.sh

```diff
@@ -1,40 +1,74 @@
-#!/bin/bash
+#!/usr/bin/env bash
+
+# -e Stops execution in the instance of a command or pipeline error
+# -u Treat unset variables as an error and exit immediately
+set -eu
+
 # Run all tests and coverage checks. Called from Travis automatically, also
 # suitable to run manually. See list of prerequisite packages in .travis.yml
 if type realpath >/dev/null 2>&1 ; then
-  cd $(realpath $(dirname $0))
+  cd "$(realpath -- $(dirname -- "$0"))"
 fi
 
-set -exo pipefail
-TRAVIS=${TRAVIS:-false}
+#
+# Defaults
+#
+STATUS="FAILURE"
+# RUN, UNIT_FILTER, and INT_FILTER vars are here for backwards compatibility,
+# and should be set to () unconditionally in the future.
+if [ -z "${RUN+x}" ]; then RUN=(); fi
+if [ -z "${UNIT_FILTER+x}" ]; then UNIT_FILTER=(); fi
+if [ -z "${INT_FILTER+x}" ]; then INT_FILTER=(); fi
+ENABLE_UNIT_RACE="false"
+export BOULDER_CONFIG_DIR="${BOULDER_CONFIG_DIR:-"test/config"}"
 
-# The list of segments to run. To run only some of these segments, pre-set the
-# RUN variable with the ones you want (see .travis.yml for an example).
-# Order doesn't matter. Note: gomod-vendor is specifically left out of the
-# defaults, because we don't want to run it locally (it could delete local
-# state) We also omit coverage by default on local runs because it generates
-# artifacts on disk that aren't needed.
-RUN=${RUN:-lints unit integration}
+#
+# Print Functions
+#
+function print_outcome() {
+  if [ "$STATUS" == SUCCESS ]
+  then
+    echo -e "\e[32m"$STATUS"\e[0m"
+  else
+    echo -e "\e[31m"$STATUS"\e[0m"
+  fi
+}
+
+function print_list_of_integration_tests() {
+  for file in ./test/integration/*.go; do
+    [ -e "$file" ] || continue
+    cat "$file" | grep -e '^func Test' | awk '{print $2}' | sed s/\(t//
+  done
+  exit 0
+}
+
+function exit_msg() { echo "$*" >&2; exit 2; }  # complain to STDERR and exit with error
+function check_arg() { if [ -z "$OPTARG" ]; then exit_msg "No arg for --$OPT option, use: -h for help">&2; fi; }
+function trap_outcome_on_exit() { trap "print_outcome" EXIT; }
```

Comment:
This is called in only one place; I think it would be simpler to just write the `trap` call inline instead of defining a function for it.

Reply:
I am good with changing it, though, I found that it wasn't clear enough to the reader what the statement was actually doing, the function name allows the reader to almost immediately recognize, even without a comment, what we're doing when this is called.

Reply:
I'm always a big fan of using variable names and function names to make things clearer. In this case, my feeling is that if someone doesn't know what `trap "print_outcome" EXIT` means, they're not any more likely to know what `trap_outcome_on_exit` means.